x86: minowboard: Enable userspace mode

AdithyaBaglody opened the pull request.

AdithyaBaglody committed:
x86: minnowboard: Add support for enabling MMU

The required linker and gen_mmu.py support added for minnowboard.

Signed-off-by: Adithya Baglody <adithya.nagaraj.baglody@intel.com>

AdithyaBaglody committed:
x86: minnowboard: Enable the userspace mode

This enables the MMU with PAE mode and also enables the userspace
mode.

Signed-off-by: Adithya Baglody <adithya.nagaraj.baglody@intel.com>

andrewboie commented on include/arch/x86/linker.ld:
```diff
@@ -159,6 +159,9 @@ SECTIONS
 #ifdef CONFIG_APPLICATION_MEMORY
 	SECTION_DATA_PROLOGUE(_APP_DATA_SECTION_NAME, (OPTIONAL),)
 	{
+#ifndef CONFIG_XIP
```
why was this change made? should be in its own commit

AdithyaBaglody committed:
x86: linker: Maintain 4K alignment for application memory.

When MMU is enabled and the SOC we are running doesn't have
execute in-place(XIP) the final image will be a monolith which
sits in RAM. In such situations we need to maintain the alignment
for application memory. If not maintained the MMU boot tables
will not be configured properly.

Signed-off-by: Adithya Baglody <adithya.nagaraj.baglody@intel.com>
